src/pkg/mapperclient/client.go

```diff
@@ -142,35 +145,40 @@ func (c *Client) Intents(ctx context.Context, namespaces []string, labels []stri
 	return res.Intents, nil
 }
 
-func (c *Client) ListIntents(ctx context.Context, namespaces []string, withLabelsFilter bool, labels []string, withKafkaIntents bool) ([]IntentsIntentsIntent, error) {
-	if !withLabelsFilter && !withKafkaIntents {
-		serviceIntents, err := c.ServiceIntents(ctx, namespaces)
-		if err != nil {
-			return nil, err
-		}
-
-		return serviceIntentsToIntents(serviceIntents), nil
-	} else if !withKafkaIntents {
-		serviceIntentsWithLabels, err := c.ServiceIntentsWithLabels(ctx, namespaces, labels)
-		if httpErr := (HTTPError{}); errors.As(err, &httpErr) && httpErr.StatusCode == http.StatusUnprocessableEntity {
-			err = errors.New("listing intents with labels filter is not supported by your network mapper. Please upgrade")
-		}
-		if err != nil {
-			return nil, err
-		}
-
-		return serviceIntentsWithLabelsToIntents(serviceIntentsWithLabels), nil
-	} else {
+func (c *Client) ListIntents(ctx context.Context, namespaces []string, withLabelsFilter bool, labels []string) ([]IntentsIntentsIntent, error) {
+	if withLabelsFilter {
 		intents, err := c.Intents(ctx, namespaces, labels)
 		if httpErr := (HTTPError{}); errors.As(err, &httpErr) && httpErr.StatusCode == http.StatusUnprocessableEntity {
-			err = errors.New("listing intents including kafka intents is not supported by your network mapper. Please upgrade")
+			logrus.Warnf("Using an old network mapper version. A newer version is available " +
+				"which includes Kafka topics & HTTP resources for Istio. Please upgrade: https://github.com/otterize/network-mapper")
+
+			serviceIntentsWithLabels, err := c.ServiceIntentsWithLabels(ctx, namespaces, labels)
+			if httpErr := (HTTPError{}); errors.As(err, &httpErr) && httpErr.StatusCode == http.StatusUnprocessableEntity {
+				err = errors.New("listing intents with labels filter is not supported by your network mapper. Please upgrade")
+			}
+			if err != nil {
+				return nil, err
+			}
+
+			return serviceIntentsWithLabelsToIntents(serviceIntentsWithLabels), nil
 		}
+		return intents, nil
+	}
+
+	intents, err := c.Intents(ctx, namespaces, labels)
+	if httpErr := (HTTPError{}); errors.As(err, &httpErr) && httpErr.StatusCode == http.StatusUnprocessableEntity {
+		logrus.Warnf("Using an old network mapper version. A newer version is available " +
+			"which includes Kafka topics & HTTP resources for Istio. Please upgrade: https://github.com/otterize/network-mapper")
```

Comment:
Can you change 'please upgrade' to 'please deploy a new version of network mapper to use these new features'? The upgrade + link to github suggests that you will be able to upgrade there, but most likely you need to update your chart instead